src/language-server/project/rover/DocumentSynchronization.ts

```diff
@@ -109,10 +123,12 @@ export class DocumentSynchronization {
     }
   };
 
-  private async sendDocumentChanges(document: TextDocument) {
+  private async sendDocumentChanges(
+    document: TextDocument,
+    previousParts = this.knownFiles.get(document.uri)?.parts || [],
```

Comment:
`previousParts` can now be passed as an argument, as in the case of a restart of the upstream LSP we want to re-send the full document as a series of `didOpen` notifications using this function.